absl/strings/str_cat_test.cc

```diff
@@ -408,6 +408,18 @@ TEST(StrCat, VectorBoolReferenceTypes) {
   EXPECT_EQ(result, "1010");
 }
 
+TEST(StrCat, UBCheck) {
```

Comment:
Can you give this test case a more descriptive name than just `UBCheck`? Maybe `AvoidsMemcpyWithNullptr`? You should also put a comment explaining that passing `nullptr` to `memcpy` is undefined behavior and that test is to catch that bug.

Reply:
Sure, fixed. I also added ABSL_PREDICT_TRUE to mitigate possible performance degradation.